ui/src/components/VaultSummary.jsx

```diff
@@ -94,13 +93,15 @@ export function VaultSummary({ vault, brandToInfo, id }) {
   } = makeDisplayFunctions(brandToInfo);
 
   const {
-    debt, // amount
+    debtSnapshot, // amount
     interestRate, // ratio
     liquidationRatio, // ratio
     locked, // amount
     status, // string
   } = vault;
 
+  const { debt = null } = debtSnapshot ?? {};
```

Comment:
Done thanks.

I'm not too eager to jump to TypeScript. It's probably more likely to pay off the more complex a dapp becomes, but not sure if it's worth rewriting this one at this point. It's sometimes too tempting to slap a bunch of `any`s on things especially when dealing with external libraries or browser APIs, so it requires a certain amount of commitment to get value out of it. I could probably be talked into it with some pertinent examples, but I don't feel as though types have been a limiting factor in productivity so far.

It would definitely be good to have some example dapp template that uses TypeScript, but also important that we maintain some templates using JavaScript.

Reply:
That's cool. Though there wouldn't be any rewriting. We'd just change the extension of some files to `.ts` and then be able to use that syntax. We aren't doing that in most of SDK because it would entail a build step, but JSX already requires a build step.

Reply:
Ack. As discussed I think we want to look into typescript at some point soon